cpp/src/rolling/rolling.cu

```diff
@@ -724,6 +746,30 @@ std::unique_ptr<column> rolling_window(column_view const& input,
   }
 }
 
+namespace 
+{
+  using sort_groupby_helper = cudf::experimental::groupby::detail::sort::sort_groupby_helper;
+  using index_vector = sort_groupby_helper::index_vector;
+
+  std::pair<index_vector, index_vector> get_group_offsets_and_labels(
+    table_view const& group_keys, 
+    column_view const& input) 
+    {
+      if (group_keys.num_columns() > 0) {
+        // Groupby columns are available.
+        sort_groupby_helper helper{group_keys, cudf::include_nulls::YES, cudf::sorted::YES}; 
+        return std::make_pair(helper.group_offsets(), helper.group_labels());
+      }
+      else {
+        // No Groupby columns specified. Treat as one big group.
+        return std::make_pair(
+          index_vector{std::vector<size_type>{0, input.size()}},
+          index_vector{std::vector<size_type>(input.size(), 0)}
```

Comment:
Isn't this causing the same behaviour as your change mentioned here: https://github.com/rapidsai/cudf/pull/4363#discussion_r394092612. If so, that method would be preferred for performance reasons. This way right here, you make a host allocation the size of your input and then need a H->D memcpy.

Reply:
This should now be fixed now, in commit e9ec50b0166b3b3fca64d2bfe4460d48a2b2dd51. I thought this might be unavoidable in the `time_range` case, but that should now be sorted out as well.